medical/models/medical_patient.py

```diff
@@ -95,6 +95,15 @@ def action_invalidate(self):
             ('z', 'law marriage'),
         ])
 
+    @api.multi
+    def onchange_state(self, state_id):
+        '''
+        Hack to allow for onchange_state on partner.
+        Uses current user's partner, as there should be no correlation
+        between the partner and the response from this method anyways
+        '''
+        return self.env.user.partner_id.onchange_state(state_id)
```

Comment:
Thanks @pedrobaeza - In this instance, it works out well due to the simplicity of the parent method & the fact that the logic for it basically could never change. 

This really kind of hits at my engineering bone though, because we're re-implementing a method from another module. Is there not a more elegant strategy to completely reimplenting?

That said, I'll definitely make the change here - I highly doubt Odoo is going to change the name of the `country_id` col, so it's safe IMO. Just wondering for future.